pandas/core/groupby.py

```diff
@@ -997,16 +997,16 @@ def tail(self, n=5):
         Examples
         --------
 
-        >>> df = DataFrame([[1, 2], [1, 4], [5, 6]],
+        >>> df = DataFrame([['a', 1], ['a', 2], ['b', 1], ['b', 2]],
                             columns=['A', 'B'])
         >>> df.groupby('A', as_index=False).tail(1)
            A  B
-        0  1  2
-        2  5  6
+        0  a  2
+        2  b  2
```

Comment:
@kidphys head and filter are regarded as 'filter' operations, and not 'aggregation' operations, so in that case `as_index` is ignored. So, it is not a bug, but it also shouldn't be in the docs, as this can only cause confusion that it has no effect.